include/nuttx/power/battery_gauge.h

```diff
@@ -93,31 +93,32 @@ struct battery_gauge_operations_s
 {
   /* Return the current battery state (see enum battery_status_e) */
 
-  int (*state)(struct battery_gauge_dev_s *dev, int *status);
+  int (*state)(FAR struct battery_gauge_dev_s *dev, FAR int *status);
```

Comment:
Please add the `CODE` keyword before each function pointer definition, reference here:
https://github.com/apache/nuttx/blob/master/include/nuttx/fs/fs.h#L203-L232

Reply:
Maybe should explain what is `segmented memory` and why it is important those addresses been accessible outside their respective segments.

Reply:
> Please add the `CODE` keyword before each function pointer definition, reference here: https://github.com/apache/nuttx/blob/master/include/nuttx/fs/fs.h#L203-L232

why https://github.com/apache/nuttx/blob/48c153db25a4db852ea601116ee9309fc3a90754/include/nuttx/fs/fs.h#L223 miss CODE？

Reply:
Far/near/code pointers were used in old platforms like DOS, NuttX is still supporting this segmented memory platfrom architecture, like 16-bit x86 architecture, we need to ensure that the compatibility is not broken.